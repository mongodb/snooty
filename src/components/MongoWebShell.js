import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/react';

const SOURCE_URL = 'https://mws.mongodb.com/';

const MongoWebShell = ({
  nodeData: {
    options: { version },
  },
}) => {
  return (
    <iframe
      css={css`
        border: 0;
      `}
      title="MongoDB Web Shell"
      allowFullScreen
      sandbox="allow-scripts allow-same-origin"
      width="100%"
      height="320"
      src={version ? `${SOURCE_URL}?version=${version}` : SOURCE_URL}
    />
  );
};

MongoWebShell.propTypes = {
  nodeData: PropTypes.shape({
    options: PropTypes.shape({
      version: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default MongoWebShell;
