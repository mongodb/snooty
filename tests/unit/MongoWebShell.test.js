import React from 'react';
import { render } from '@testing-library/react';
import MongoWebShell from '../../src/components/MongoWebShell';

// data for this component
import mockDataDefault from './data/MongoWebShell-default.test.json';
import mockDataVersioned from './data/MongoWebShell-v4.2.test.json';

it('correctly renders a MongoDB Web Shell', () => {
  const tree = render(<MongoWebShell nodeData={mockDataDefault} />);
  expect(tree.asFragment()).toMatchSnapshot();
});

it('correctly renders a MongoDB Web Shell v4.2.9', () => {
  const tree = render(<MongoWebShell nodeData={mockDataVersioned} />);
  expect(tree.asFragment()).toMatchSnapshot();
});
