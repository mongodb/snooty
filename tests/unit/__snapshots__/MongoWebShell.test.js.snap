// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders a MongoDB Web Shell 1`] = `
.emotion-0 {
  border: 0;
}

<iframe
  allowfullscreen=""
  class="emotion-0"
  height="320"
  sandbox="allow-scripts allow-same-origin"
  src="https://mws.mongodb.com/"
  title="MongoDB Web Shell"
  width="100%"
/>
`;

exports[`correctly renders a MongoDB Web Shell v4.2.9 1`] = `
.emotion-0 {
  border: 0;
}

<iframe
  allowfullscreen=""
  class="emotion-0"
  height="320"
  sandbox="allow-scripts allow-same-origin"
  src="https://mws.mongodb.com/?version=4.2"
  title="MongoDB Web Shell"
  width="100%"
/>
`;
